blueoil/datasets/fer_2013.py

```diff
@@ -74,46 +74,43 @@ def num_per_epoch(self):
 
         return len(self.images)
 
-    def _load_data(self, path):
+    def _load_data(self, path, data_type):
         # Load the label and image data from csv
         lines = _load_csv(path)
-        train = []
-        public_test = []
-        for line in lines:
-            if line[2] == "Training":
-                train.append(line)
-            elif line[2] == "PublicTest":
-                public_test.append(line)
-
-        train_num = len(train)
-        public_test_num = len(public_test)
-        image_size = self.image_size
 
-        train_images = np.empty((train_num, image_size, image_size), dtype=np.uint8)
-        train_labels = np.empty(train_num, dtype=np.uint8)
-        public_test_images = np.empty((public_test_num, image_size, image_size), dtype=np.uint8)
-        public_test_labels = np.empty(public_test_num, dtype=np.uint8)
+        target = ""
+        if data_type == "train":
+            target = "Training"
+        elif data_type == "validation":
+            target = "PublicTest"
+        elif data_type == "test":
+            target = "PrivateTest"
+        else:
+            raise ValueError("Must provide data_type = train or validation or test")
+
+        data = [line for line in lines if line[2] == target]
+        data_num = len(data)
+        image_size = self.image_size
 
-        for i in range(train_num):
-            train_images[i] = np.array(train[i][1].split(' '), np.uint8).reshape((image_size, image_size))
-            train_labels[i] = train[i][0]
-        for i in range(public_test_num):
-            public_test_images[i] = np.array(public_test[i][1].split(' '), np.uint8).reshape((image_size, image_size))
-            public_test_labels[i] = public_test[i][0]
+        images = np.empty((data_num, image_size, image_size, 3), dtype=np.uint8)
+        labels = np.empty(data_num, dtype=np.uint8)
+        for i in range(data_num):
+            tmp_img = np.array(data[i][1].split(' '), np.uint8).reshape((image_size, image_size))
+            # convert grayscale to RGB
```

Comment:
Please add comments why we need this conversion.
And I think this operation dose not convert gratscale image to RGB, just expand grayscale image's channel size.